xcp_d/utils/utils.py

```diff
@@ -119,125 +119,130 @@ def get_transformfilex(bold_file, mni_to_t1w, t1w_to_native):
-        transformfileMNI = [str(mni6c_to_t1w), str(t1w_to_mni)]
-        transformfileT1W = [str(mni6c_to_t1w)]
-
-    elif 'space-MNIInfant' in file_base:
-        transformfileMNI = str(
-            pkgrf('xcp_d', 'data/transform/infant_to_2009_Composite.h5'))
-        transformfileT1W = str(mni_to_t1w)
-
-    elif 'space-MNIPediatricAsym' in file_base:
-        mnisf = mni_to_t1w.split('from-')[0]
-        mni6c_to_t1w = glob.glob(
-            mnisf + 'from-MNIPediatricAsym*_to-T1w_mode-image_xfm.h5')[0]
-        t1w_to_mni = glob.glob(mnisf + 'from-T1w_to-' + template
-                               + '*_mode-image_xfm.h5')[0]
-        transformfileMNI = [str(mni6c_to_t1w), str(t1w_to_mni)]
-        transformfileT1W = [str(mni6c_to_t1w)]
-
-    elif 'space-T1w' in file_base:
-        mnisf = mni_to_t1w.split('from')[0]
-        transformfileMNI = [str(t1w_to_mni)]
-        transformfileT1W = [
-            str(pkgrf('xcp_d', 'data/transform/oneratiotransform.txt'))
-        ]
+    # Extract the space of the BOLD file
+    file_base = os.path.basename(bold_file)
+    bold_space = re.findall("space-([a-zA-Z0-9]+)", file_base)
+    if not len(bold_space):
+        bold_space = "native"
+    else:
+        bold_space = bold_space[0]
+
+    if bold_space in ("native", "T1w"):
+        base_std_space = re.findall("from-([a-zA-Z0-9]+)", mni_to_t1w)[0]
+    elif f"from-{bold_space}" not in mni_to_t1w:
+        raise ValueError(f"Transform does not match BOLD space: {bold_space} != {mni_to_t1w}")
+
+    # Pull out the correct transforms based on bold_file name and string them together.
+    xforms_to_T1w = [mni_to_t1w]  # used for all spaces except T1w and native
+    xforms_to_T1w_invert = [False]
+    if bold_space == "MNI152NLin2009cAsym":
+        # Data already in MNI152NLin2009cAsym space.
+        xforms_to_MNI = ["identity"]
+        xforms_to_MNI_invert = [False]
+
+    elif bold_space == "MNI152NLin6Asym":
+        # MNI152NLin6Asym --> MNI152NLin2009cAsym
+        MNI152NLin6Asym_to_MNI152NLin2009cAsym = str(
+            get_template(
+                template="MNI152NLin2009cAsym",
+                mode="image",
+                suffix="xfm",
+                extension=".h5",
+                **{"from": "MNI152NLin6Asym"},
+            ),
+        )
+        xforms_to_MNI = [MNI152NLin6Asym_to_MNI152NLin2009cAsym]
+        xforms_to_MNI_invert = [False]
+
+    elif bold_space == "MNIInfant":
+        # MNIInfant --> MNI152NLin2009cAsym
+        MNIInfant_to_MNI152NLin2009cAsym = pkgrf(
+            "xcp_d",
+            "data/transform/tpl-MNIInfant_from-MNI152NLin2009cAsym_mode-image_xfm.h5",
+        )
+        xforms_to_MNI = [MNIInfant_to_MNI152NLin2009cAsym]
+        xforms_to_MNI_invert = [False]
+
+    elif bold_space == "T1w":
+        # T1w --> ?? (extract from mni_to_t1w) --> MNI152NLin2009cAsym
+        # Should not be reachable, since xcpd doesn't support T1w-space BOLD inputs
```

Comment:
Is this true?

Reply:
We only allow standard space (MNI152NLin2009cAsym, MNI152NLin6Asym, MNIInfant) inputs at the moment. I just kept those elements in the code in case we want to support T1w or native-space BOLD data in the future, since it might be preferable to concatenate the xforms from native to the target space.